### PR TITLE
[WiP] Add Makefile

### PR DESCRIPTION
**Pull request by lkollar**
This implements parts of the release process outlined in #396. I'm not sure of the exact commands used in the release checklist so I've improvised in a few places, but I intended this as a proof of concept and definitely needs mork work.

Let me know what steps are wrong/missing.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+.PHONY: test-upload
+test-upload: SETUP_EXTRA_ARGS = -r https://test.pypi.org/legacy/
+test-upload: upload
+
+.PHONY: release
+release: test fetch_tzmeta sdist bdist_wheel
+
+.PHONY: clean
+clean:
+	-rm -rf *.egg-info
```
Comment (pganssle) [inline Makefile]: `__pycache__`?

**Comment by pganssle** [comment]
This doesn't seem like a significant improvement over my `release.py` script, IMO.

**Comment by lkollar** [comment]
I had no idea about `release.py`. The issue linked mentions a checklist only.

**Comment by pganssle** [comment]
I probably wrote `release.py` after I wrote that "automated release procedure" issue, or around the same time.

It's basically the same as this script, except written in Python, which I find a bit easier to maintain. The downsides are:

1. It's non-standard (unlike `make`), so discoverability is probably reasonably low.
2. It does not run the tests against the built libraries.

I *could* also do a sort of automated release where the CI builds the package, runs the tests, and delivers them to PyPI. That's not out of the question, but I don't *love* the trust model there.

**Comment by lkollar** [comment]
I had no idea that parts of the release process was already automated. It doesn't make much sense to have a Makefile and the release script as well so I can just close this PR.

#396 should probably be closed or updated with the missing release steps?

**Pull request closed without merging**